### ISSUE TITLE
AWS 2 Lambda native support

### ISSUE THREAD

**Issue by ppalaga** [apache/camel-quarkus#1572]

**Comment by Jeansen** [comment]
I think this one is quite simple to do, I'd give it a try this or next weekend.

**Comment by ppalaga** [comment]
That's highly welcome, unless @oscerd has something in the works?

**Comment by oscerd** [comment]
Hello @Jeansen 

Feel free to work on this extension :-)

Thanks a lot for your time!

**Comment by Jeansen** [comment]
I am sorry, this the passed WE I did not find time. I hope this WE works out. This time, it will - sadly - not be done in 9 days ;-)

**Comment by Jeansen** [comment]
When promoting to native, I got some error regarding the POM.xml file in the integration-tests folder

```
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project  (/home/marcel/ownCloud/workspaces/projects/camel-quarkus/integration-tests/pom.xml) has 1 error
[ERROR]     Non-parseable POM /home/marcel/ownCloud/workspaces/projects/camel-quarkus/integration-tests/pom.xml: Duplicated tag: 'modules' (position: START_TAG seen ...</dependencyManagement>\n\n    <modules>... @53:14)  @ line 53, column 14 -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/ModelParseException
```

And in the POM, I see

```
 <modules>
        <module>aws2-lambda</module>
    </modules>

    <properties>
        <quarkus.banner.enabled>false</quarkus.banner.enabled>
    </properties>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>io.quarkus</groupId>
                <artifactId>quarkus-junit5-internal</artifactId>
                <version>${quarkus.version}</version>
            </dependency>
        </dependencies>
    </dependencyManagement>

    <modules>
        <!-- build those first -->
        <module>core</module>
        <module>main</module>
        <module>main-devmode</module>
        <module>main-xml-jaxb</module>
        <module>main-xml-io</module>
        <module>main-collector</module>
        <module>main-caffeine-lrucache</module>
        <module>main-command-mode</module>
.
.
.
```

Obviously, something with the module addition went wrong ;-) I'll fix it manually for now.

**Comment by Jeansen** [comment]
As I see, integration tests are all in one folder. So, I assume I do not have to amend the 'tooling/scripts/test-categories.yaml' file?
Is there any automation? Or should I just manually throw away what I do not need from the generated integration test and simply try adding a rout the the aws2 test class?

**Comment by oscerd** [comment]
You should add just the needed bits into the camel-aws2 test folder. It should be enough to add the extension and write a similar route.

**Comment by Jeansen** [comment]
OK, thx.

**Comment by ppalaga** [comment]
> Obviously, something with the module addition went wrong

Thanks for complaining :), let me check that tomorrow.

**Comment by ppalaga** [comment]
> > Obviously, something with the module addition went wrong
> 
> Thanks for complaining :), let me check that tomorrow.

Should be fixed via f4233ff8159e15f4ba2ad46bc16be6ba754c317e
